### ISSUE TITLE
Library does not work with Epson LS11000 (and probably LS12000) 

### ISSUE THREAD

**Issue by mag1024** [pszafer/epson_projector#16]
It seems that Epson chose to disable the HTTP interface on their newer models, and so the default HA integration doesn't work at all. I hacked it to use the TCP mode and that somewhat works, but there are a few other weird things going on, that I don't fully understand. Like why does the serial fetch happen on a different port than the rest of the TCP connection? My projector does not have it open:
```
nmap -p 0-5000 projector
Starting Nmap 7.80 ( https://nmap.org ) at 2022-07-16 00:14 EDT
Nmap scan report for projector (10.x.x.x)
Host is up (0.00092s latency).
Not shown: 4999 closed ports
PORT     STATE SERVICE
3629/tcp open  escvpnet
4352/tcp open  pjlink
```

It seems to respond fine on the regular port (3629) to the SNO command, even when power is off -- so that check seems unnecessary as well.

**Comment by pszafer** [comment]
Hi,
maybe it's just disabled? You need to enable "web remote" to open port 80.
3620 is standard port for TCP connection when projector is **ON**
```
  HTTP port   (port   number   used   for   Web   Control   except   for80   (default),   843,   3620,   3621,   3625,   3629,   4352,   4649,   5357,10000,   10001,   41794
```

Did you turn on projector to check for open ports?
Serial number is possible only to fetch when projector is ON and for most projector only on port 3620.

**Comment by mag1024** [comment]
All the available network options I could find are turned on:
![network](https://user-images.githubusercontent.com/787978/179366058-b6586d00-c380-4d10-8c3e-b686b6ab7378.JPG)

The set of open ports does not change whether the projector is on or off, and SNO? works fine on 3629 even when the projector is the power state "04" -- which is listed as Standby / Network On, so that makes sense.

**Comment by mag1024** [comment]
FWIW, I have a working fork/rewrite using only TCP, port 3629, and IMEVENT-based push instead of polling, as suggested in #13, at  [mag1024/epson-projector-modern-homeassistant](https://github.com/mag1024/epson-projector-modern-homeassistant) -- but it only supports a very limited set of functionality for now: power and input selection.

**Comment by pszafer** [comment]
Nice, I think best way would be to add functionality of yours system to my library and you'd need to maintain custom component for now (and maybe add later to Core HA)

**Comment by molusk** [comment]
I confirm I have the same issue with Epson LS12000b.
It is not listening to port 80 no matter what but tcp/3629 is available when the projector is on.
It seems to disconnect from the network when off...

**Comment by pszafer** [comment]
Maybe there is Energy Saving on?

**Comment by molusk** [comment]
Yes, I have found an option which seems to disable network connection when off but not in Energy saving options. 
Operations, Standby mode => Communication On  
I enabled it but did not check since...

Anyway, as there is no service listening on tcp/80, this integration won't work. I will test @mag1024's fork soon.

**Comment by DazWorrall** [comment]
@mag1024 what is the status of your fork? Do you need testers?

**Comment by molusk** [comment]
@DazWorrall I use it on a daily basis since my last comment. It works perfectly and has all the (basic) functions I need 👍

**Comment by DazWorrall** [comment]
It's working well for me too! Thanks.

**Comment by amosyuen** [comment]
I wrote a TCP serial home assistant integration on HACS for my Epson Home Cinema that some people may find useful https://github.com/amosyuen/ha-epson-projector-link.

**Comment by molusk** [comment]
Hi @amosyuen, 
I installed your integration too, it works nicely and was not being disconnected as @mag1024 one did (but I think he corrected something as it did not happen for a long time now).
Both are very similar for simple use (on/off, HDMI source).

I was expecting to be able to switch lens and picture memory as some yaml files on the repo suggest it is possible but the *_memory services are not available. Furthermore, I would need to know what signal the projector gets to automatically switch picture memory bank (between HDR and SDR) and it does not seem available either. (By the way, this function should be directly integrated in the firmware, Epson, if you read this 😸 )
